ArticleTemplates/assets/js/modules/ophan-iframe.js

```diff
@@ -0,0 +1,20 @@
+const handleIframeMessage = (messageEvent) => {
+    const correctType = messageEvent.data.type === 'ophan-iframe-click-event';
+    const correctOrigin = messageEvent.origin === 'https://www.theguardian.com' ||
+        messageEvent.origin === 'https://m.code.dev-theguardian.com/';
+
+    if (correctOrigin && correctType) {
+        const clickEvent = messageEvent.data.value;
+        if (window.GuardianJSInterface && window.GuardianJSInterface.trackInPageClick) {
+            window.GuardianJSInterface.trackInPageClick(JSON.stringify(clickEvent));
+        }
+    }
+};
+
+const init = () => {
+    window.top.addEventListener('message', handleIframeMessage, false);
```

Comment:
Do you need `.top` here? Isn't this code already running in the topmost frame?

Reply:
I guess not? I'll change that :)